hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -337,30 +337,15 @@ protected void initMetadataTable(WriteOperationType operationType, Option<String
    *
    * @param inFlightInstantTimestamp - The in-flight action responsible for the metadata table initialization
    */
-  private void initializeMetadataTable(WriteOperationType operationType, Option<String> inFlightInstantTimestamp) {
+  private void initializeMetadataTable(Option<String> inFlightInstantTimestamp) {
     if (!config.isMetadataTableEnabled()) {
       return;
     }
 
     try (HoodieTableMetadataWriter writer = SparkHoodieBackedTableMetadataWriter.create(context.getHadoopConf().get(), config,
-            context, Option.empty(), inFlightInstantTimestamp)) {
+        context, Option.empty(), inFlightInstantTimestamp)) {
       if (writer.isInitialized()) {
```

Comment:
We added the guard rail for table services in MDT to be triggered only by regular writers in Data table, so that for a single writer modes with async table services, there won't be any race conditions. 
Ref: https://github.com/apache/hudi/pull/3900
But the code evolved and we automatically enable in process lock provider for single writer mode w/ async table services. And so, we should be good to remove the constraint. Just that we might keep triggering the schedule of compaction in MDT everytime. May be we can intercept from the active timeline on when is the last time, compaction was triggered and add in some optimization. 
Nothing required for this patch. But as a follow up.